### PR TITLE
Aggregator Publishing to Smart contract

### PR DESCRIPTION
**Pull request by hmzakhalid**
## Summary by CodeRabbit

- **New Features**
	- Expanded event handling capabilities with new event types for adding and removing cipher nodes.
	- Introduced the `EvmCaller` module for managing interactions with Ethereum contracts.
	- Implemented methods for publishing plaintext outputs and committees to Ethereum smart contracts.
	- Added a new command-line argument for the registry filter contract.
	- Enhanced the script to dynamically handle public key values.
	- Added a new environment variable for the registry filter contract in the integration tests.

- **Bug Fixes**
	- Enhanced error handling capabilities within the enclave event management.

- **Documentation**
	- Expanded test coverage for event handling and connectivity features of the `MainAggregator` and `MainCiphernode`.

### CODE REVIEW

**Review by ryardley — DISMISSED** [review]
🚀 looks good to me

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ name: "INTEGRATION"
 env:
   HARDHAT_VAR_MNEMONIC: "test test test test test test test test test test test junk"
   HARDHAT_VAR_INFURA_API_KEY: "zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz"
+  PRIVATE_KEY: "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"
```
Comment (auryn-macmillan) [inline .github/workflows/integration.yml:6]: Not necessary right now, but we should move all of these to environmental variables at some point, just so we or others don't accidentally put sensitive information here.

**Review by auryn-macmillan — APPROVED** [review]
lgtm

File: .github/workflows/integration.yml (hunk 1 above)
Comment (hmzakhalid) [reply .github/workflows/integration.yml:6]: Agreed, this PK specifically is for the integration tests to pass as it needs to publish the PubKey and Decrypted PT

**Pull request merged**